client/src/partials/templates/modals/dateInterval.js

```diff
@@ -0,0 +1,69 @@
+angular.module('bhima.controllers')
+.controller('DateIntervalModalController', DateIntervalModalController);
+
+DateIntervalModalController.$inject = ['$uibModalInstance'];
+
+/**
+ * Date Interval Modal
+ * This controller permit to select Interval of date
+ * it returns the result as dateFrom and dateTo
+ */
+function DateIntervalModalController (Instance) {
+  var vm = this;
+
+  vm.options = [
+    { label : 'FORM.LABELS.TODAY', fn : day },
+    { label : 'FORM.LABELS.THIS_WEEK', fn : week },
+    { label : 'FORM.LABELS.THIS_MONTH', fn : month },
+    { label : 'FORM.LABELS.OTHER', fn : other },
+  ];
+
+  // expose to the viewe
+  vm.search = search;
+  vm.submit = submit;
+  vm.close  = close;
+
+  // start up the modal
+  startup();
+
+  // submit
+  function submit() {
+    Instance.close({ dateFrom: vm.dateFrom, dateTo: vm.dateTo });
+  }
+
+  // close
+  function close() {
+    Instance.dismiss();
+  }
+
+  function search (selection) {
+    vm.selected = selection.label;
+    selection.fn();
+  }
+
+  function day () {
+    vm.dateFrom = new Date();
+    vm.dateTo = new Date();
+  }
+
+  function week () {
```

Comment:
I would also propose that you look at [this function](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/partials/patients/registry/search.modal.js#L64) for setting the date range.